_rules/aria-required-context-role-ff89c9.md

```diff
@@ -0,0 +1,257 @@
+---
+id: ff89c9
+name: ARIA required context role
+rule_type: atomic
+description: | 
+  This rule checks that a role exists inside its required context.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects: 
+- DOM Tree
+- CSS Styling
+authors:
+- Anne Thyme Nørregaard
+---
+
+## Applicability
+
+The rule applies to any HTML or SVG element that is [included in the accessibility tree](#included-in-the-accessibility-tree) and has an explicit [semantic role](#semantic-role) with a [WAI-ARIA required context role](https://www.w3.org/TR/wai-aria-1.1/#scope), except if the element has an implicit semantic role that is identical to its explicit semantic role.
+
+**Note:** An example of an element that has an implicit semantic role that is identical to its explicit semantic role is an `<li>` element that has `role="listitem"`. This rule is not applicable to elements like this.
+
+## Expectation
+
+The target element is [owned by](#owned-by) an element that has a [semantic role](#semantic-role) that is one of the [WAI-ARIA required context roles](https://www.w3.org/TR/wai-aria-1.1/#scope) of the target element.
+
+**Note:** The definition of [owned by](#owned-by) used in this rule is diverging from the definition of ["owned element" in WAI-ARIA](https://www.w3.org/TR/wai-aria-1.1/#dfn-owned-element). See more in the [owned by](#owned-by) definition.
+
+## Assumptions
+
+This rule assumes that the explicit [semantic role](#semantic-role) on the target element is used with the intention to comply to WCAG. If the explicit semantic role on the target element is incorrectly used, and any relationships between elements are already programmatically determinable, failing this rule might not result in accessibility issues for users of assistive technologies, and it should then not be considered a failure under WCAG success criterion 1.3.1 Info and Relationships.
+
+## Accessibility Support
+
+- This rule relies on assistive technologies to recognize which elements are [owned by](#owned-by) each other. This includes when the element is owned by another element that is an ancestor, but not a parent of the target element. Some assistive technologies do not accept these owned by relationships, unless workarounds are used.
+Furthermore, `aria-owns` has limited support in some user agents.
+- Some user agents and assistive technologies ignore empty elements, which means they are not presented to all users. However, since this is handled inconsitently across user agents and assistive technologies, this rule is applicable to empty elements. 
+
+## Background
+
+- [Required Context Role](https://www.w3.org/TR/wai-aria-1.1/#scope)
```

Comment:
Missing a link to understanding, also are there any techniques?

Reply:
Link to Understanding added. No techniques map in a sensible way.